superset-frontend/src/components/Datasource/DatasourceEditor.jsx

```diff
@@ -1078,14 +1103,60 @@ class DatasourceEditor extends PureComponent {
                       <TextAreaControl
                         language="sql"
                         offerEditInModal={false}
-                        minLines={20}
+                        minLines={10}
                         maxLines={Infinity}
                         readOnly={!this.state.isEditMode}
                         resize="both"
                         tooltipOptions={sqlTooltipOptions}
                       />
                     }
+                    additionalControl={
+                      <div
+                        css={css`
+                          position: absolute;
+                          right: 0;
+                          top: 0;
+                          z-index: 2;
+                        `}
+                      >
+                        <Button
+                          css={css`
+                            align-self: flex-end;
+                            height: 24px;
+                            padding-left: 6px;
+                            padding-right: 6px;
+                          `}
+                          size="small"
+                          buttonStyle="primary"
+                          onClick={() => {
+                            this.onQueryRun();
+                          }}
+                        >
+                          <Icons.CaretRightFilled
```

Comment:
I think for consistency it would be better to have a button that says "Run", like in SQL Lab (no need for the dropdown options):

<img width="290" alt="Screenshot 2025-05-08 at 10 21 23 AM" src="https://github.com/user-attachments/assets/9fb9439d-03c2-41f8-8cda-91f9f66601ab" />

Then, in the future we could add the limit dropdown as well, it could be useful.